# Урок 5. Практическое задание
## Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование

## Задание 1
### Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16.
### Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

Воспользуемся формулами

$$
    \overline{X} - z_{\alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}}
    ,    
    \overline{X} + z_{\alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}}
$$

    a = 1 - 0.95 = 0.05
    Определяем z по даблице для значения 0,975 (1 - a / 2)
    z = 1.96

In [17]:
m = 80
n = 256
p = 0.95
s = 16
a = 0.05 # 1 - p
z = 1.96 # из таблицы

print(f'Доверительный интервал: {m - (z * s / n ** 0.5), m + (z * s / n ** 0.5)}')

Доверительный интервал: (78.04, 81.96)


## Задание 2
### В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1
### Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

Воспользуемся формулами

$$
    \overline{X} - t_{\alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}}
    ,    
    \overline{X} + t_{\alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}}
$$

    a = 1 - 0.95 = 0.05
    Определяем критерий Стьюдента t по таблице для 
    v = n - 1 = 9 
    a / 2 = 0.05 / 2 = 0.025 
    t = 2.262

In [35]:
import numpy as np

x = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
m = x.mean()
s = x.std(ddof=1)
n = 10
t = 2.262 # из таблицы

print(f'Доверительный интервал: {m - (t * s / n ** 0.5):.4}, {m + (t * s / n ** 0.5):.4}')

Доверительный интервал: 6.268, 6.912


## Задание 3
### Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
### Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

    Н0: диаметр = 17 мм, µ = µ0
    Н1: диаметр = 17,5 мм, µ > µ0

    Используем критерий Z–тест 
$$
    Z_{H} = \left( \overline{X} - \mu_{0}\right) \cdot \dfrac{\sqrt{n}}{\sigma}
$$

    Z определяем по таблице для а - 0.05
    Z = 1.645

In [34]:
m_0 = 17
m_1 = 17.5
n = 100
d = 4
a = 0.05 
z_0 = 1.645 # из таблицы

s = d ** 0.5
z_1 = (m_1 - m_0) * (n ** 0.5) / s

print(f'Zh = {z_1}, Z = 1.645')

Zh = 2.5, Z = 1.645


#### Т.к. Zh > Z (2.5 > 1.645) гипотеза H0 не верна, принимаем гипотезу H1

## Задание 4
### Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. 
### Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

    Н0: вес пачки = 200 г., µ = µ0
    Н1: вес пачки <> 200 г., µ <> µ0

Воспользуемся формулами

$$
    \overline{X} - t_{\alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}}
    ,    
    \overline{X} + t_{\alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}}
$$

    a = 1 - 0.99 = 0.01
    Определяем критерий Стьюдента t по таблице для 
    v = n - 1 = 9 
    a / 2 = 0.01 / 2 = 0.005 
    t = 3.250
    
Найдем доверительный интервал

In [38]:
import numpy as np

x = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
m = x.mean()
s = x.std(ddof=1)
n = 10
t = 3.250 # из таблицы

print(f'Доверительный интервал: {m - (t * s / n ** 0.5):.4}, {m + (t * s / n ** 0.5):.4}')

Доверительный интервал: 193.9, 203.1


#### Утверждение продавца верно т.к. вес 200 г. входит в доверительный диапазон (193.9 > 200 > 203.1) и гипотеза H0 верна